README.md

```diff
@@ -80,6 +83,24 @@ Rough outline of a possible resolution algorithm:
 
 In the future, the algorithm might be adjusted to align with work done in the [import maps proposal](https://github.com/domenic/import-maps).
 
+### Validations and Fallbacks
+
+The following validations are performed for an exports start to resolve:
+
+- The exports target must be a string and start with _"./"_ (URLs and absolute paths are not currently supported but may be supported in future).
+- The exports target cannot backtrack below the package base.
+- Exports targets cannot map into a nested node_modules path.
+
+For directory resolutions the following validations also apply:
+
+- Directory exports targets must end in a trailing _"/"_.
+- Directory exports targets may not backtrack below the package base.
```

Comment:
typically .. goes up, and into a directory goes down, i think?
```suggestion
- Directory exports targets may not backtrack above the package base.
```

Reply:
I think this language assumes a string where going back (and going up) is "going left". Up/down vs. back/forward. Either works I assume.

Reply:
I like to think of the filesystem as a Tree.

<img src="https://user-images.githubusercontent.com/229881/62880122-8fef5100-bce1-11e9-85c7-c1301db3a5e8.png" width=200 />

Each time you open a directory, you are going "down" the tree closer to the leaf nodes.

You walk "up" the tree to get to the root.

Reply:
tbh I’ve never heard of a filesystem mentally modeled as anything else but a tree.

Reply:
Agreed! Although confusingly the analogy of a Tree in CS already turns real trees upside down 🙃. Or maybe it's been roots and not leaves all along :P